pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -147,25 +162,7 @@ mkDerivation (finalAttrs: {
 
   inherit src;
 
-  patches = patches fetchpatch ++ lib.optionals withNativeCompilation [
-    (substituteAll {
-      src = if lib.versionOlder finalAttrs.version "29"
-            then ./native-comp-driver-options-28.patch
-            else if lib.versionOlder finalAttrs.version "30"
-            then ./native-comp-driver-options.patch
-            else ./native-comp-driver-options-30.patch;
-      backendPath = (lib.concatStringsSep " "
-        (builtins.map (x: ''"-B${x}"'') ([
-          # Paths necessary so the JIT compiler finds its libraries:
-          "${lib.getLib libgccjit}/lib"
-        ] ++ libGccJitLibraryPaths ++ [
-          # Executable paths necessary for compilation (ld, as):
-          "${lib.getBin stdenv.cc.cc}/bin"
-          "${lib.getBin stdenv.cc.bintools}/bin"
-          "${lib.getBin stdenv.cc.bintools.bintools}/bin"
-        ])));
-    })
-  ];
+  patches = patches fetchpatch ++ lib.optionals withNativeCompilation [ nativeCompilationPatch ];
```

Comment:
Factoring out `substituteAll` seems over-engineering to me.  It also makes the diff larger than needed and harder to review.

Reply:
Hmm, formatting makes the diff larger than needed and harder to review.  Could you keep the old format?  We can do formatting in a future PR or leave it to the tree-wide one.